tests/search/top_terms_collector_test.cpp

```diff
@@ -205,7 +205,7 @@ struct sub_reader final : irs::SubReader {
     info.live_docs_count = num_docs;
   }
 
-  virtual uint64_t CountMappedMemory() const { return 0; }
+  uint64_t CountMappedMemory() const final { return 0; }
```

Comment:
Firstly try final, if necessary override, and only for first declaration (and ofc only if necessary) virtual.
Because of such rules, there was possible to remove final from a lot of class-es, and make managed_ptr sizeof 1 pointer instead of 2